README
Print Math::BigInt's as named numbers in different styles

(C) ..--- ----- ----- .---- by - . .-.. ... .-.-.-  All rights reserved.

This package is a subclass of Math::BigInt and was inspired by Chris London
Noll's 'number.pl' program.

To test, unzip & untar package, then run

	perl Makefile.PL
	make
	make test
	sudo make install

See ./examples/number.pl for how to use this package on a webserver or on the
command line.

See INSTALL for detailed help on installing these modules.

Please send me test-reports, your experiences with this and your ideas - I love
to hear about my work!

Tels <http://bloodgate.com/>


CHANGES
0.06 2021-07-16

 * Add support for Norwegian.

 * Better argument processing and checking.

 * Fix typo in English name ('twelf' -> 'twelve').

0.05 2021-07-14

 * Reorganize existing test files, and add more test files.

 * The tests now require Test::More version 0.88 or newer.

 * New MANIFEST.SKIP file based on the template in the ExtUtils-Manifest
   distribution.

 * Format CHANGES according to CPAN::Changes::Spec.

0.04 2016-12-03 pjacklam

 * Use ExtUtils::MakeMaker rather than Module::Install in Makefile.PL

 * Remove Build.PL. It is no longer supported.

 * Fix subclassing in Math::BigInt::Named, i.e., replace
  "use Math::BigInt::Named" by "use Math::BigInt".

 * Remove use of the discouraged "base" package in subclasses of
   Math::BigInt::Named.

 * Better documentation in .pm files.

 * Remove author information in LICENSE file.

 * Update README file.

 * Trim whitespace.

 * Inform that the TODO file is not up to date.

0.03 2007-02-24 Tels 181 tests

 * modernize package (META.yml, SIGNATURE, etc.)

 * bundle and use Module::Install

 * add POD tests

0.02 2001-12-30 Tels

 * added German and English (own subclass), some tests and some real code.

0.01 2001-11-20 Tels

 * first release


lib/Math/BigInt/Named.pm
#!perl

package Math::BigInt::Named;

use 5.006001;
use strict;
use warnings;

use Carp qw( carp croak );

use Math::BigInt 1.97;
our @ISA = qw(Math::BigInt);

our $VERSION = '0.07';

# Globals.

our ($accuracy, $precision, $round_mode, $div_scale);
$accuracy   = undef;
$precision  = undef;
$round_mode = 'even';
$div_scale  = 40;

# Not all of them exist yet.
my $LANGUAGE = {
  en => 'english',
  de => 'german',
  sp => 'spanish',
  fr => 'french',
  ro => 'romana',
  it => 'italian',
  no => 'norwegian',
  };

# Index of languages that have been loaded.

my $LOADED = { };

sub name {
    # output the name of the number
    my $x = shift;

    # make Math::BigInt::Name -> name(123) work
    $x = $x -> new(shift) unless ref($x);

    return 'NaN' if $x -> is_nan();

    my @args = ();
    if (@_) {
        if (ref($_[0]) eq 'HASH') {
            carp "When the options are given as a hash ref, additional",
              " arguments are ignored" if @_ > 1;
            @args = %{ $_[0] };
        } else {
            @args = @_;
        }
    }

    my $lang;
    while (@args) {
        my $param = shift @args;
        croak "Parameter name can not be undefined" unless defined $param;
        croak "Parameter name can not be an empty string" unless length $param;

        if ($param =~ /^lang(uage)?$/) {
            $lang = shift @args;
            croak "Language can not be undefined" unless defined $lang;
            croak "Language can not be an empty string" unless length $lang;
            next;
        }

        croak "Invalid parameter '$param'";
    }

    $lang = 'english' unless defined $lang;
    $lang = $LANGUAGE -> {$lang} if exists $LANGUAGE -> {$lang}; # en => english

    $lang = 'Math::BigInt::Named::' . ucfirst($lang);

    if (!defined $LOADED -> {$lang}) {
        eval "require $lang;";
        croak "Can't load module '$lang'" if $@;
        $LOADED -> {$lang} = 1;
    }

    my $y = $lang -> new($x);
    $y -> name();
}

sub from_name {
    # Create a Math::BigInt::Name from a name string. Not implemented.

    my $x = Math::BigInt -> bnan();
}

1;

__END__

=pod

=head1 NAME

Math::BigInt::Named - Math::BigInts that know their name in some languages

=head1 SYNOPSIS

    use Math::BigInt::Named;

    $x = Math::BigInt::Named->new("123");

    print $x->name(),"\n";                      # default is english
    print $x->name( language => 'de' ),"\n";    # but German is possible
    print $x->name( language => 'German' ),"\n";        # like this
    print $x->name( { language => 'en' } ),"\n";        # this works, too

    print Math::BigInt::Named->from_name("einhundert dreiundzwanzig"),"\n";

=head1 DESCRIPTION

This is a subclass of Math::BigInt and adds support for named numbers.

=head1 METHODS

=head2 name()

    print Math::BigInt::Name->name( 123 );

Convert a BigInt to a name.

=head2 from_name()

    my $bigint = Math::BigInt::Name->from_name('hundertzwanzig');

Create a Math::BigInt::Name from a name string. B<Not yet implemented!>

=head1 BUGS

Please report any bugs or feature requests to
C<bug-math-bigint-named at rt.cpan.org>, or through the web interface at
L<https://rt.cpan.org/Ticket/Create.html?Queue=Math-BigInt-Named>
(requires login).
We will be notified, and then you'll automatically be notified of progress on
your bug as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Math::BigInt::Named

You can also look for information at:

=over 4

=item * GitHub

L<https://github.com/pjacklam/p5-Math-BigInt-Named>

=item * RT: CPAN's request tracker

L<https://rt.cpan.org/Dist/Display.html?Name=Math-BigInt-Named>

=item * MetaCPAN

L<https://metacpan.org/release/Math-BigInt-Named>

=item * CPAN Testers Matrix

L<http://matrix.cpantesters.org/?dist=Math-BigInt-Named>

=item * CPAN Ratings

L<https://cpanratings.perl.org/dist/Math-BigInt-Named>

=back

=head1 LICENSE

This program is free software; you may redistribute it and/or modify it under
the same terms as Perl itself.

=head1 SEE ALSO

L<Math::BigInt> and L<Math::BigFloat>.

=head1 AUTHORS

=over 4

=item *

(C) by Tels http://bloodgate.com in late 2001, early 2002, 2007.

=item *

Maintainted by Peter John Acklam E<lt>pjacklam@gmail.comE<gt>, 2016-.

=item *

Based on work by Chris London Noll.

=back

=cut


lib/Math/BigInt/Named/German.pm
#!perl

package Math::BigInt::Named::German;

use 5.006001;
use strict;
use warnings;

use Math::BigInt::Named;
our @ISA = qw< Math::BigInt::Named >;

our $VERSION = '0.06';

sub name
  {
  # output the name of the number
  my ($x) = shift;
  $x = Math::BigInt->new($x) unless ref($x);

  my $self = ref($x);

  return '' if $x->is_nan();

  my $index = 0;

  my $ret = '';
  my $y = $x->copy(); my $rem;
  if ($y->sign() eq '-')
    {
    $ret = 'minus ';
    $y->babs();
    }
  if ($y < 1000)
    {
    return $ret . $self->_triple($y,1,0);
    }
  while (!$y->is_zero())
    {
    ($y,$rem) = $y->bdiv(1000);
    $ret = $self->_triple($rem,0,$index)
          .' ' . $self->_triple_name($index,$rem) . ' ' . $ret;
    $index++;
    }
  $ret =~ s/\s+$//;	# trailing spaces
  $ret;
  }

my $SMALL = [ qw/
  null
  eins
  zwei
  drei
  vier
  fuenf
  sechs
  sieben
  acht
  neun
  zehn
  oelf
  zwoelf
  dreizehn
  vierzehn
  fuenfzehn
  sechzehn
  siebzehn
  achtzehn
  neunzehn
  / ];

my $ZEHN = [ qw /
  zehn
  zwanzig
  dreissig
  vierzig
  fuenfzig
  sechzig
  siebzig
  achtzig
  neunzig
  / ];

my $HUNDERT = [ qw /
  ein
  zwei
  drei
  vier
  fuenf
  sechs
  sieben
  acht
  neun
  / ];

my $TRIPLE = [ qw /
  mi
  bi
  tri
  quadri
  penti
  hexi
  septi
  octi
  / ];

sub _triple_name
  {
  my ($self,$index,$number) = @_;

  return '' if $index == 0 || $number->is_zero();
  return 'tausend' if $index == 1;

  my $postfix = 'llion'; my $plural = 'en';
  if ($index & 1 == 1)
    {
    $postfix = 'lliarde'; $plural = 'n';
    }
  $postfix .= $plural unless $number->is_one();
  $index -= 2;
  $TRIPLE->[$index >> 1] . $postfix;
  }

sub _triple
  {
  # return name of a triple (aka >= 0, and <= 1000)
  # input: number 	>= 0, < 1000)
  #        only    	true if triple is the only triple ever ($nr < 1000)
  #	   index	0 for last triple, 1 for tausend, 2 for million etc
  my ($self,$number,$only,$index) = @_;

  # eins, ein hundert, ein tausend, eine million
  # zwei, zwei hundert, zwei tausend, zwei million

  my $eins = 'ein';
  $eins = 'eins' if $index == 0;
  $eins = 'eine' if $index > 2;

  return '' if $number->is_zero() && !$only;	# 0 => null, but only for one
  return $eins if $number->is_one();
  return $SMALL->[$number] if $number < scalar @$SMALL;	# known name

  my $hundert = $number / 100;
  my $rem = $number % 100;
  my $rc = '';
  $rc = "$HUNDERT->[$hundert-1]hundert" if !$hundert->is_zero();

  my $concat = ''; $concat = 'und' if $rc ne '';
  return $rc if $rem->is_zero();
  return $rc . $concat . $SMALL->[$rem] if $rem < scalar @$SMALL;

  my $zehn; ($zehn,$rem) = $rem->bdiv(10);

  $rc .= $concat . $HUNDERT->[$rem-1] if !$rem->is_zero(); # 31, 32..
  $concat = ''; $concat = 'und' if $rc ne '';
  $rc .= $concat . $ZEHN->[$zehn-1] if !$zehn->is_zero();  # 1,2,3..

  $rc;
  }

1;

__END__

=pod

=head1 NAME

Math::BigInt::Named::German - Math::BigInts that know their name in German

=head1 SYNOPSIS

  use Math::BigInt::Named::German;

  $x = Math::BigInt::Named::German->new($str);

  print $x->name(),"\n";

  print Math::BigInt::Named::German->from_name('einhundert dreiundzwanzig),"\n";

=head1 DESCRIPTION

This is a subclass of Math::BigInt and adds support for named numbers
with their name in German to Math::BigInt::Named.

Usually you do not need to use this directly, but rather go via
L<Math::BigInt::Named>.

=head1 METHODS

=head2 name()

	print Math::BigInt::Name->name( 123 );

Convert a BigInt to a name.

=head2 from_name()

	my $bigint = Math::BigInt::Name->from_name('hundertzwanzig');

Create a Math::BigInt::Name from a name string. B<Not yet implemented!>

=head1 BUGS

For information about bugs and how to report them, see the BUGS section in the
documentation available with the perldoc command.

    perldoc Math::BigInt::Named

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Math::BigInt::Named::English

For more information, see the SUPPORT section in the documentation available
with the perldoc command.

    perldoc Math::BigInt::Named

=head1 LICENSE

This program is free software; you may redistribute it and/or modify it under
the same terms as Perl itself.

=head1 SEE ALSO

L<Math::BigInt::Named>, L<Math::BigInt> and L<Math::BigFloat>.

=head1 AUTHORS

=over 4

=item *

(C) by Tels http://bloodgate.com in late 2001, early 2002, 2007.

=item *

Maintained by Peter John Acklam E<lt>pjacklam@gmail.comE<gt>, 2016-.

=back

=cut


lib/Math/BigInt/Named/English.pm
#!perl

package Math::BigInt::Named::English;

use 5.006001;
use strict;
use warnings;

use Math::BigInt::Named;
our @ISA = qw< Math::BigInt::Named >;

our $VERSION = '0.07';

sub name {
    my $x = shift;
    $x = Math::BigInt -> new($x) unless ref($x);

    my $class = ref($x);

    return '' if $x -> is_nan();

    my $index = 0;

    my $ret = '';
    my $y = $x -> copy();
    my $rem;

    if ($y -> sign() eq '-') {
        $ret = 'minus ';
        $y -> babs();
    }

    if ($y < 1000) {
        return $ret . $class -> _triple($y, 1, 0);
    }

    while (!$y -> is_zero()) {
        ($y, $rem) = $y -> bdiv(1000);
        $ret = $class -> _triple($rem, 0, $index)
          . ' ' . $class -> _triple_name($index, $rem) . ' ' . $ret;
        $index++;
    }
    $ret =~ s/\s+$//;           # trailing spaces
    $ret;
}

my $SMALL = [ qw/
  zero
  one
  two
  three
  four
  five
  six
  seven
  eight
  nine
  ten
  eleven
  twelve
  thirteen
  fourteen
  fifteen
  sixteen
  seventeen
  eighteen
  nineteen
  / ];

my $ZEHN = [ qw /
  ten
  twenty
  thirty
  fourty
  fifty
  sixty
  seventy
  eighty
  ninety
  / ];

my $HUNDERT = [ qw /
  one
  two
  three
  four
  five
  six
  seven
  eight
  nine
  / ];

my $TRIPLE = [ qw /
  mi
  bi
  tri
  quadri
  penti
  hexi
  septi
  octi
  / ];

sub _triple_name {
    my ($self, $index, $number) = @_;

    return '' if $index == 0 || $number -> is_zero();
    return 'thousand' if $index == 1;

    my $postfix = 'llion';
    my $plural = 's';
    if (($index & 1) == 1) {
        $postfix = 'lliard';
    }
    $postfix .= $plural unless $number -> is_one();
    $index -= 2;
    $TRIPLE -> [$index >> 1] . $postfix;
}

sub _triple {
    # return name of a triple (aka >= 0, and <= 1000)
    # input: number     >= 0, < 1000
    #        only       true if triple is the only triple ever ($nr < 1000)
    #       index       0 for last triple, 1 for thousand, 2 for million etc
    my ($self, $number, $only) = @_;

    return '' if $number -> is_zero() && !$only; # 0 => null, but only for one
    return $SMALL -> [$number] if $number < scalar @$SMALL; # known name

    my $hundert = $number / 100;
    my $rem = $number % 100;
    my $rc = '';
    $rc = $HUNDERT -> [$hundert-1] . "hundred" if !$hundert -> is_zero();

    my $concat = '';
    $concat = 'and' if $rc ne '';
    return $rc if $rem -> is_zero();
    return $rc . $concat . $SMALL -> [$rem] if $rem < scalar @$SMALL;

    my $zehn;
    ($zehn, $rem) = $rem -> bdiv(10);

    my $last = '';
    $last = $HUNDERT -> [$rem-1] if !$rem -> is_zero();    # 31, 32..
    $last = $ZEHN -> [$zehn-1] . $last if !$zehn -> is_zero(); # 1, 2, 3..

    $rc . $last;
}

1;

__END__

=pod

=head1 NAME

Math::BigInt::Named::English - Math::BigInts that know their name in English

=head1 SYNOPSIS

    use Math::BigInt::Named::English;

    $x = Math::BigInt::Named::English->new($str);

    print $x->name(),"\n";

    print Math::BigInt::Named::English->from_name('one hundredandfive),"\n";

=head1 DESCRIPTION

This is a subclass of Math::BigInt and adds support for named numbers
with their name in English to Math::BigInt::Named.

Usually you do not need to use this directly, but rather go via
L<Math::BigInt::Named>.

=head1 METHODS

=head2 name()

    print Math::BigInt::Name->name( 123 );

Convert a BigInt to a name.

=head2 from_name()

    my $bigint = Math::BigInt::Name->from_name('hundertzwanzig');

Create a Math::BigInt::Name from a name string. B<Not yet implemented!>

=head1 BUGS

For information about bugs and how to report them, see the BUGS section in the
documentation available with the perldoc command.

    perldoc Math::BigInt::Named

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Math::BigInt::Named::English

For more information, see the SUPPORT section in the documentation available
with the perldoc command.

    perldoc Math::BigInt::Named

=head1 LICENSE

This program is free software; you may redistribute it and/or modify it under
the same terms as Perl itself.

=head1 SEE ALSO

L<Math::BigInt::Named>, L<Math::BigInt> and L<Math::BigFloat>.

=head1 AUTHORS

=over 4

=item *

(C) by Tels http://bloodgate.com in late 2001, early 2002, 2007.

=item *

Maintained by Peter John Acklam E<lt>pjacklam@gmail.comE<gt>, 2016-.

=back

=cut


lib/Math/BigInt/Named/Norwegian.pm
#!perl

package Math::BigInt::Named::Norwegian;

use strict;
use warnings;

use Math::BigInt::Named;
our @ISA = qw< Math::BigInt::Named >;

our $VERSION = '0.06';

my $SMALL = [ qw/
  null
  en
  to
  tre
  fire
  fem
  seks
  syv
  åtte
  ni
  ti
  elleve
  tolv
  tretten
  fjorten
  femten
  seksten
  sytten
  atten
  nitten
  / ];

my $TENS = [ qw /
  ti
  tjue
  tretti
  førti
  femti
  seksti
  sytti
  åtti
  nitti
  / ];

my $HUNDREDS = [ qw /
  en
  to
  tre
  fire
  fem
  seks
  sju
  åtte
  ni
  / ];

my $TRIPLE = [ qw /
  mi
  bi
  tri
  kvadri
  kvinti
  seksti
  septi
  okti
  noni
  desi
  undesi
  duodesi
  tredesi
  kvattordesi
  kvindesi
  seksdesi
  septendesi
  oktodesi
  novemdesi
  viginti
  / ];

sub name {
    my $x = shift;
    $x = Math::BigInt -> new($x) unless ref($x);

    my $class = ref($x);

    return '' if $x -> is_nan();

    my $ret = '';
    my $y = $x -> copy();
    my $rem;

    if ($y -> sign() eq '-') {
        $ret = 'minus ';
        $y -> babs();
    }

    if ($y < 1000) {
        return $ret . $class -> _triple($y, 1, 0);
    }

    # Split the number into numerical triplets.

    my @num = ();
    while (!$y -> is_zero()) {
        ($y, $rem) = $y -> bdiv(1000);
        unshift @num, $rem;
    }

    # Convert each numerical triplet into a string.

    my @str = ();
    for my $i (0 .. $#num) {
        my $num = $num[$i];
        my $str;
        my $index = $#num - $i;

        my $count;
        if ($num == 1 && $index == 1) {
            $count = "ett";     # "ett tusen", not "en tusen"
        } else {
            $count = $class -> _triple($num, 0, $i);
        }
        $str .= $count;

        # "tusen", "million"/"millioner", "milliard/milliarder", ...

        if ($index > 0) {
            my $triple_name = $class -> _triple_name($#num - $i, $num);
            $str .= ' ' . $triple_name;
        }

        $str[$i] = $str;
    }

    # 1100 -> "ett tusen ett hundre"    (not "ett tusen og ett hundre")
    # 1099 -> "ett tusen og nittini"    (not "ett tusen nittini")
    # 1098 -> "ett tusen og nittiåtte"  (not "ett tusen nittiåtte")
    # ...
    # 1001 -> "ett tusen og en"         (not "ett tusen en")
    # 1000 -> "ett tusen"               (not "ett tusen og null"

    if (@num > 1 && 0 < $num[-1] && $num[-1] < 100) {
        splice @str, -1, 0, "og";
    }

    $ret . join(" ", grep /\S/, @str);
}

sub _triple_name {
    my ($self, $index, $number) = @_;
    # index => 0 hundreds, tens and ones
    # index => 1 thousands
    # index => 2 millions

    return '' if $index == 0 || $number -> is_zero();
    return 'tusen' if $index == 1;

    my $postfix = 'llion';
    my $plural = 'er';
    if (($index & 1) == 1) {
        $postfix = 'lliard';
    }
    $postfix .= $plural unless $number -> is_one();
    $index -= 2;
    $TRIPLE -> [$index >> 1] . $postfix;
}

sub _triple {
    # return name of a triple
    # input: number     >= 0, < 1000
    #        only       true if triple is the only triple
    my ($self, $number, $only) = @_;

    # 0 => null, but only if there is just one triple
    return '' if $number -> is_zero() && !$only;

    # we have the full name for these
    return $SMALL -> [$number] if $number <= $#$SMALL;

    # New code:

    my @num = ();
    $num[1] = $number % 100;                # tens and ones
    $num[0] = ($number - $num[1]) / 100;    # hundreds

    my @str = ();

    # Do the hundreds, if any.

    if ($num[0]) {
        my $str;
        $str = $num[0] == 1 ? "ett"     # "ett hundre", not "en hundre"
                            : $HUNDREDS -> [$num[0] - 1];
        $str .= " hundre";
        push @str, $str;
    }

    # Do the tens and ones, if any.

    if ($num[1]) {
        my $str;
        my $ones = $num[1] % 10;
        my $tens = ($num[1] - $ones) / 10;
        if ($num[1] <= $#$SMALL) {
            $str = $SMALL -> [ $num[1] ];
        } else {
            $str = $TENS -> [ $tens - 1];
            if ($ones > 0) {
                $str .= "";
                $str .= $SMALL -> [ $ones ];
            }
        }
        push @str, $str;
    }

    return join " og ", @str;
}

1;

__END__

=pod

=head1 NAME

Math::BigInt::Named::Norwegian - Math::BigInts that know their name in Norwegian

=head1 SYNOPSIS

    use Math::BigInt::Named::Norwegian;

    $x = Math::BigInt::Named::Norwegian -> new("123");
    $str = $x -> name();

    $str = "ett hundre og to";
    $x = Math::BigInt::Named::Norwegian -> from_name($str);

=head1 DESCRIPTION

This is a subclass of Math::BigInt and adds support for named numbers
with their name in Norwegian to Math::BigInt::Named.

Usually you do not need to use this directly, but rather go via
L<Math::BigInt::Named>.

=head1 METHODS

=head2 name()

    print Math::BigInt::Name -> name( 123 );

Convert a BigInt to a name.

=head2 from_name()

    my $bigint = Math::BigInt::Name -> from_name('ett hundre og to');

Create a Math::BigInt::Name from a name string. B<Not yet implemented!>

=head1 BUGS

For information about bugs and how to report them, see the BUGS section in the
documentation available with the perldoc command.

    perldoc Math::BigInt::Named

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Math::BigInt::Named::Norwegian

For more information, see the SUPPORT section in the documentation available
with the perldoc command.

    perldoc Math::BigInt::Named

=head1 LICENSE

This program is free software; you may redistribute it and/or modify it under
the same terms as Perl itself.

=head1 SEE ALSO

L<Math::BigInt::Named> and L<Math::BigInt>.

=head1 AUTHORS

=over 4

=item *

Peter John Acklam E<lt>pjacklam@gmail.comE<gt>, 2021.

=back

=cut


Makefile.PL
# -*- mode: perl -*-

use strict;
use warnings;

use ExtUtils::MakeMaker;

WriteMakefile
  (
   'NAME'               => 'Math::BigInt::Named',
   'VERSION_FROM'       => 'lib/Math/BigInt/Named.pm',
   'AUTHOR'             => 'Peter John Acklam <pjacklam@gmail.com>',
   'MIN_PERL_VERSION'   => 5.006001,

   # A hash of modules that are required to run Makefile.PL itself, but not to
   # run the distribution.

   'CONFIGURE_REQUIRES' => {
                            'ExtUtils::MakeMaker' => 6.58,
                           },

   # A hash of modules that are needed to build the module but not run it.

   'BUILD_REQUIRES'     => {},

   # A hash of modules that are needed to run the module.

   'PREREQ_PM'          => {
                            'Math::BigInt'        => 1.97,
                           },

   # A hash of modules that are needed to test the module but not run or build
   # it.

   'TEST_REQUIRES'      => {
                            'Test::More'          => 0.88,
                           },

   'LICENSE'            => 'perl_5',
   'LIBS'               => [''],        # e.g., '-lm'
   'DEFINE'             => '',          # e.g., '-DHAVE_SOMETHING'
   'INC'                => '',          # e.g., '-I/usr/include/other'
   'SIGN'               => 1,

   'META_MERGE'         => {
       'meta-spec' => { version => 2 },
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/pjacklam/p5-Math-BigInt-Named.git',
                web  => 'https://github.com/pjacklam/p5-Math-BigInt-Named',
            },
        },
    },
  );
